src/core/elf.c

```diff
@@ -4,9 +4,15 @@
 #include <string.h>
 
 int is_elf(elf_header_t* elf);
-void load_segment(uint64_t* data, elf_program_header_t* program_header);
-
-elf_header_t* elf_load(uint64_t* data) {
+void load_segment(uint8_t* data, elf_program_header_t* program_header);
+inline elf_section_header_t* get_elf_section_header(elf_header_t* header) __attribute__((
```

Comment:
I saw your comment about the perf boost but I would get rid of the `inline`/`__attribute__((always_inline))` for now anyway. It is going to introduce some complexity that we likely don't need right now. I'd rather file an issue to keep the idea somewhere.

Reply:
How do you mean? It simply turns the functions into inline code, it won't cause any complexity. If it really bothers you so much, I can change it, though.

My argument for keeping it is that we'll use these functions a lot and we want to be as fast as possible, so it'd make sense to keep them.

Reply:
I dont think this is entirely correct. `inline` has different meanings depending on the compiler, C version, etc. (https://stackoverflow.com/questions/216510/what-does-extern-inline-do)

Compilers are smart so I think speed optimizations will still happen, even without this keyword (https://stackoverflow.com/questions/1759300/when-should-i-write-the-keyword-inline-for-a-function-method)

That's why I suggest to drop it.

Reply:
Well, it's late and I don't really have the brain power to get into a big discussion over it. I removed the inlines, as you suggested 😄